drivers/hmis/app/graphql/types/hmis_schema/enrollment.rb

```diff
@@ -36,124 +42,134 @@ def self.configuration
     end
 
     description 'HUD Enrollment'
+
+    # LIMITED ACCESS FIELDS. These fields are visible with `can_view_limited_enrollment_details` permission (or `can_view_enrollment_details` + `can_view_project` perms).
```

Comment:
The `Enrollment` GraphQL type is now basically "split" - the majority of the fields require `can_view_enrollment_details` permissions. 

The downside with this approach is that it is difficult to maintain. If someone adds a new field as a `field` or without specifying the required permission, it would be inadvertently exposed to users who should only have Limited access. (They wouldn't see the data in the application, but they could modify a query to fetch additional data).

I think another approach would be to resolve a union type - `LimitedEnrollment` and `Enrollment`. I looked into this a little bit but it seemed somewhat difficult to do with our pagination class. Happy to look into that further, or other ideas/patterns, if folks have any ideas.

Reply:
What about adding a new super method to use instead of `field` have `field` default to the more safe path, but to expose something to the limited users you'd have to call `limited_field` or something.  That way you have to be more intentional when exposing  data.

Reply:
Looks like you took the opposite approach and `detail_field` is the less safe path?  I'm a bit confused on what is being limited in this case, maybe a different name like "summary" or "public" would be easier to reason about?

Reply:
Ok, updated with https://github.com/greenriver/hmis-warehouse/pull/3682/commits/bfa0210d1b48581854bb1c080b3bd24eae54bd97. Much cleaner I think.